setuptools/config/_apply_pyprojecttoml.py

```diff
@@ -30,14 +30,14 @@
 from ..warnings import SetuptoolsWarning
 
 if TYPE_CHECKING:
-    from distutils.dist import _OptionsList
+    from distutils.dist import _OptionsList  # Comes from typeshed
     from setuptools._importlib import metadata
     from setuptools.dist import Distribution
     from typing_extensions import TypeAlias
 
 EMPTY: Mapping = MappingProxyType({})  # Immutable dict-like
 _ProjectReadmeValue: TypeAlias = Union[str, Dict[str, str]]
-_CorrespFn: TypeAlias = Callable[["Distribution", Any, StrPath], None]
+_CorrespFn: TypeAlias = Callable[["Distribution", Any, Union[StrPath, None]], None]
```

Comment:
Do we need the optional StrPath here? I think it is better to keep `StrPath` instead of `Union[StrPath, None]`, no?

In the `apply` function the very first thing we do is:

`root_dir = os.path.diname(filename) or "."`, so we always have a `StrPath` value that we use to call al the functions, not?

Reply:
We don't *need* to, it would simply be arbitrary restrictions since all correspondances in `PYPROJECT_CORRESPONDENCE` *do* support a `None` `root_dir` (they all end up defaulting to `os.getcwd()` or `os.curdir`)